examples/test_messaging/test_messaging.c

```diff
@@ -0,0 +1,398 @@
+                                        printf("---------------------------\n");
+                                        msg_params->tests_failed++;
+                                        msg_params->test_phase++;
+                                }
+                        }
+
+                }
+                
+                rte_free(msg_data);
+
+                msg_params->test_msg_count = 0;
+
+                printf("TEST 2: Send/Receive Multiple Messages...\n");
+                printf("---------------------------\n");
+
+                for(int i = 0; i < 10; i++){
+                        int* msg_int = (int*)rte_malloc(NULL, sizeof(int), 0);
+                        *msg_int = i;    
+                        onvm_nflib_send_msg_to_nf(msg_params->address, msg_int);
+                }
+
+        }
+        //Tests if multiple messages can be sent to itself
+        else if(msg_params->test_phase == 2){
+                
+                if(*((int *)msg_data) != msg_params->test_msg_count) { 
+                        printf("FAILED TEST 2: received %d instead of %d\n", *((int *)msg_data), msg_params->test_msg_count);
+                        printf("---------------------------\n");
+                        msg_params->tests_failed++;
+                        msg_params->test_phase++;
+                }
+                else {
+                        msg_params->test_msg_count++;
+                }
+
+                rte_free(msg_data);
+
+                if(10 == msg_params->test_msg_count){
+
+                        if(rte_ring_count(msg_params->msg_q) != 0) {
+                                printf("FAILED TEST 2: Shouldn't be any messages left, but there are %d in the ring. This may cause future tests to fail.\n", rte_ring_count(msg_params->msg_q));
+                                printf("---------------------------\n");
+                                while(rte_ring_count(msg_params->msg_q) != 0){
+                                        rte_ring_dequeue(msg_params->msg_q, msg_data);
+                                }
+                                msg_params->tests_failed++;
+                                msg_params->test_phase++;
+                        }
+                        else {
+                                printf("PASSED TEST 2\n");
+                                printf("---------------------------\n");
+                                msg_params->tests_passed++;
+                                msg_params->test_phase++;
+                                msg_params->test_msg_count = 0;
+                        }
+
+                }
+                
+                if(msg_params->test_msg_count == 0){
+                        printf("TEST 3: Message Ring Overflow...\n");
+                        printf("---------------------------\n");
+                        for(int i = 0; i < 130; i++){
```

Comment:
130 is a "magic number" that we assume is larger than the ring buffer size, but it is possible in the future we might change the size.  Can you get this number programmatically -- there is probably an API to check the size of a ring and then you could add a few extra